MovieQuiz/Services/QuestionFactory.swift

```diff
@@ -8,58 +8,117 @@
-        QuizQuestion(
-            image: "Old",
-            text: "Рейтинг этого фильма больше чем 6?",
-            correctAnswer: false),
-        QuizQuestion(
-            image: "The Ice Age Adventures of Buck Wild",
-            text: "Рейтинг этого фильма больше чем 6?",
-            correctAnswer: false),
-        QuizQuestion(
-            image: "Tesla",
-            text: "Рейтинг этого фильма больше чем 6?",
-            correctAnswer: false),
-        QuizQuestion(
-            image: "Vivarium",
-            text: "Рейтинг этого фильма больше чем 6?",
-            correctAnswer: false)
-    ]
+    private var movies: [MostPopularMovie] = []
     
+    //    private let questions: [QuizQuestion] = [
+    //        QuizQuestion(
+    //            image: "The Godfather",
+    //            text: "Рейтинг этого фильма больше чем 6?",
+    //            correctAnswer: true),
+    //        QuizQuestion(
+    //            image: "The Dark Knight",
+    //            text: "Рейтинг этого фильма больше чем 6?",
+    //            correctAnswer: true),
+    //        QuizQuestion(
+    //            image: "Kill Bill",
+    //            text: "Рейтинг этого фильма больше чем 6?",
+    //            correctAnswer: true),
+    //        QuizQuestion(
+    //            image: "The Avengers",
+    //            text: "Рейтинг этого фильма больше чем 6?",
+    //            correctAnswer: true),
+    //        QuizQuestion(
+    //            image: "Deadpool",
+    //            text: "Рейтинг этого фильма больше чем 6?",
+    //            correctAnswer: true),
+    //        QuizQuestion(
+    //            image: "The Green Knight",
+    //            text: "Рейтинг этого фильма больше чем 6?",
+    //            correctAnswer: true),
+    //        QuizQuestion(
+    //            image: "Old",
+    //            text: "Рейтинг этого фильма больше чем 6?",
+    //            correctAnswer: false),
+    //        QuizQuestion(
+    //            image: "The Ice Age Adventures of Buck Wild",
+    //            text: "Рейтинг этого фильма больше чем 6?",
+    //            correctAnswer: false),
+    //        QuizQuestion(
+    //            image: "Tesla",
+    //            text: "Рейтинг этого фильма больше чем 6?",
+    //            correctAnswer: false),
+    //        QuizQuestion(
+    //            image: "Vivarium",
+    //            text: "Рейтинг этого фильма больше чем 6?",
+    //            correctAnswer: false)
+    //    ]
+    
+    private let moviesLoader: MoviesLoading
     weak var delegate: QuestionFactoryDelegate?
     
-    init(delegate: QuestionFactoryDelegate?) {
+    //init(delegate: QuestionFactoryDelegate?) {
+    //   self.delegate = delegate
+    //}
+    
+    init(moviesLoader: MoviesLoading, delegate: QuestionFactoryDelegate?) {
+        self.moviesLoader = moviesLoader
         self.delegate = delegate
+        
+    }
+    
+    func loadData() {
+        moviesLoader.loadMovies { result in
```

Comment:
Здесь стоит добавить weak-self чтобы контекст полностью захватывался верно (в уроке могли еще не исправить)
```suggestion
        moviesLoader.loadMovies { [weak self] result in
```